app/admin/tenants/tenants.py

```diff
@@ -75,41 +76,69 @@ async def create_tenant(
-
     name = body.name
     roles = body.roles
-    if roles:
-        bound_logger.info("Onboarding `{}` with requested roles: `{}`", name, roles)
-        onboard_result = await onboard_tenant(
-            name=name,
-            roles=roles,
-            tenant_auth_token=wallet_response.token,
-            tenant_id=wallet_response.wallet_id,
-        )
 
-        bound_logger.debug("Registering actor in the trust registry")
-        await register_actor(
-            actor=Actor(
-                id=wallet_response.wallet_id,
-                name=name,
-                roles=roles,
-                did=onboard_result.did,
-                didcomm_invitation=onboard_result.didcomm_invitation,
-            )
+    actor_exists = await assert_actor_name(body.name)
+
+    if actor_exists:
+        bound_logger.info("Actor exists can't create wallet")
+        raise HTTPException(
+            409, f"Can't create Tenant. Actor with name: {name} already exists."
         )
+    else:
+        try:
+            async with get_tenant_admin_controller() as admin_controller:
+                wallet_response = await admin_controller.multitenancy.create_wallet(
+                    body=CreateWalletRequestWithGroups(
+                        image_url=body.image_url,
+                        key_management_mode="managed",
+                        label=body.name,
+                        wallet_key=base58.b58encode(token_urlsafe(48)),
+                        wallet_name=uuid4().hex,
+                        wallet_type="askar",
+                        group_id=body.group_id,
+                    )
+                )
+            bound_logger.debug("Wallet creation successful")
+
+            if roles:
+                bound_logger.info(
+                    "Onboarding `{}` with requested roles: `{}`", name, roles
+                )
+                onboard_result = await onboard_tenant(
+                    name=name,
+                    roles=roles,
+                    tenant_auth_token=wallet_response.token,
+                    tenant_id=wallet_response.wallet_id,
+                )
+                bound_logger.debug("Registering actor in the trust registry")
+                await register_actor(
+                    actor=Actor(
+                        id=wallet_response.wallet_id,
+                        name=name,
+                        roles=roles,
+                        did=onboard_result.did,
+                        didcomm_invitation=onboard_result.didcomm_invitation,
+                    )
+                )
+        except HTTPException as http_error:
+            bound_logger.info("HTTP exception: {}", http_error.detail)
+            if wallet_response:
+                bound_logger.info("delete wallet")
+                await admin_controller.multitenancy.delete_wallet(
+                    wallet_response.wallet_id
+                )
+            raise http_error from http_error
```

Comment:
This would be redundant, as `from` is used to indicate "new exception being raised is from this original exception".
For brevity, can actually just be `raise`, because that automatically raises most recently caught exception. Similar for the `raise e from e` in next catch - `e` would not need to be assigned in that case, but `http_error` is used in log of the first catch, so that can remain assigned.